File: typescript/src/neutrosophic/triplet.ts
export class Triplet {
  readonly T: number;
  readonly I: number;
  readonly F: number;

  constructor(T: number, I: number, F: number) {
    this.T = Triplet.validateScore(T, "T");
    this.I = Triplet.validateScore(I, "I");
    this.F = Triplet.validateScore(F, "F");
  }

  private static validateScore(value: number, name: string): number {
    if (typeof value !== "number" || !Number.isFinite(value)) {
      throw new TypeError(`${name} must be a finite number`);
    }

    if (value < 0 || value > 1) {
      throw new RangeError(`${name} must be between 0 and 1`);
    }

    return value;
  }
}

File: python/src/agent_squad/neutrosophic/triplet.py
from dataclasses import dataclass
from math import isfinite
from numbers import Real


@dataclass(frozen=True)
class Triplet:
    """Neutrosophic truth, indeterminacy, and falsity values."""

    T: float
    I: float
    F: float

    def __post_init__(self) -> None:
        object.__setattr__(self, "T", self._validate_component("T", self.T))
        object.__setattr__(self, "I", self._validate_component("I", self.I))
        object.__setattr__(self, "F", self._validate_component("F", self.F))

    @staticmethod
    def _validate_component(name: str, value: Real) -> float:
        if isinstance(value, bool) or not isinstance(value, Real):
            raise TypeError(f"{name} must be a number")

        numeric_value = float(value)
        if not isfinite(numeric_value):
            raise ValueError(f"{name} must be finite")
        if not 0 <= numeric_value <= 1:
            raise ValueError(f"{name} must be between 0 and 1")

        return numeric_value
